tests/test_validator.py

```diff
@@ -346,27 +346,27 @@ def test_verify_form_slots_invalid_domain(tmp_path: Path):
         forms:
           name_form:
             required_slots:
-              first_name:
-              - type: from_text
-              last_name:
-              - type: from_text
+              - first_name
+              - last_nam
         slots:
              first_name:
                 type: text
-             last_nam:
+                mappings:
+                - type: from_text
+             last_name:
                 type: text
+                mappings:
+                - type: from_text
         """
     )
     importer = RasaFileImporter(domain_path=domain)
-    validator = Validator.from_importer(importer)
     with pytest.warns(UserWarning) as w:
-        validity = validator.verify_form_slots()
```

Comment:
I've realised that by adding a new `InvalidDomain` raise in `_validate_forms()` in `domain.py` for these kind of mismatches that this validator method `verify_form_slots()`  is now redundant. Should I remove it as part of this PR?

Reply:
Does `rasa data validate` produce a warning under the same conditions as `InvalidDomain` is raised? If so yeah probably fine to get rid of it or replace it with a call to something else

Reply:
Yes, running `rasa data validate` will continue to result in error, because the domain will be empty (so the check that will fail will be [this](https://github.com/RasaHQ/rasa/blob/954107f4c6e84e506d387b65dc978452421e2acd/rasa/cli/data.py#L201)) and the warning message displayed is the one I updated this test with.

The reason why the domain is initialised as empty if `InvalidDomain` gets raised is found [here](https://github.com/RasaHQ/rasa/blob/954107f4c6e84e506d387b65dc978452421e2acd/rasa/shared/importers/rasa.py#L67).